src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -908,5 +1004,156 @@ class A
                 // error CS5001: Program does not contain a static 'Main' method suitable for an entry point
                 Diagnostic(ErrorCode.ERR_NoEntryPoint).WithLocation(1, 1));
         }
+
+        [Fact]
+        public void TaskMainAndNonTaskMain()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+class A
+{
+    static void Main()
+    {
+        System.Console.WriteLine(""Non Task Main"");
+    }
+    async static Task Main(string[] args)
+    {
+        await Task.Factory.StartNew(() => { });
+        System.Console.WriteLine(""Task Main"");
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "Non Task Main", expectedReturnCode: 0);
+        }
+
+        [Fact]
+        public void TaskMainAndNonTaskMain_CSharp71()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+class A
+{
+    static void Main()
+    {
+        System.Console.WriteLine(""Non Task Main"");
+    }
+    async static Task Main(string[] args)
+    {
+        await Task.Factory.StartNew(() => { });
+        System.Console.WriteLine(""Task Main"");
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_1)).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "Non Task Main", expectedReturnCode: 0);
+        }
+
+        [Fact]
+        public void AsyncVoidMain_CSharp7()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+class A
+{
+    async void Main(string[] args)
+    {
+        await Task.Factory.StartNew(() => { });
+        System.Console.WriteLine(""Async Void Main"");
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe).VerifyDiagnostics(
```

Comment:
>var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe).VerifyDiagnostics( [](start = 12, length = 109)

If you expect the behavior to change for the next language version, consider explicitly specifying the version that you are testing. Otherwise, the test is going to break once we bump the default version. See if other tests have the same issue as well.